File: src/main/java/src/SistemaRPG/Mago.java
package SistemaRPG;

public class Mago extends Personaje {
    private int poderMagico;
    private int danioBasico;

    public Mago(String nombre, int puntosVida, int poderMagico, int danioBasico) {
        super(nombre, puntosVida);
        this.poderMagico = poderMagico;
        this.danioBasico = danioBasico;
    }

    @Override
    public void atacar(Personaje objetivo) {
        if (poderMagico > 0) {
            System.out.println(getNombre() + " ataca físicamente a " + objetivo.getNombre() + " causando " + danioBasico + " de daño.");
            objetivo.recibirDanio(danioBasico);
        } else {
            System.out.println(getNombre() + " no tiene poder mágico para lanzar hechizos, así que ataca físicamente.");
            objetivo.recibirDanio(danioBasico);
        }
    }

    public void lanzarHechizo(Hechizo hechizo, Personaje objetivo) {
        if (hechizo.getCostoPoder() <= poderMagico) {
            objetivo.recibirDanio(hechizo.getDanio());
            poderMagico -= hechizo.getCostoPoder();
            System.out.println(getNombre() + " lanza " + hechizo.getNombre() + " a " + objetivo.getNombre() + " causando " + hechizo.getDanio() + " de daño.");
        } else {
            System.out.println(getNombre() + " no tiene suficiente poder mágico para lanzar " + hechizo.getNombre() + ".");
        }
    }

    @Override
    public String toString() {
        return super.toString() + " (Mago, Poder Mágico: " + poderMagico + ")";
    }
}

File: src/main/kotlin/SistemaRPG/Mago.kt
package SistemaRPG

class Mago(nombre: String, puntosVida: Int, private var poderMagico: Int, private val danioBasico: Int) : Personaje(nombre, puntosVida) {
    override fun atacar(objetivo: Personaje?) {
        if (poderMagico > 0) {
            println(nombre + " ataca físicamente a " + objetivo?.nombre + " causando " + danioBasico + " de daño.")
            objetivo?.recibirDanio(danioBasico)
        } else {
            println("$nombre no tiene poder mágico para lanzar hechizos, así que ataca físicamente.")
            objetivo?.recibirDanio(danioBasico)
        }
    }

    fun lanzarHechizo(hechizo: Hechizo, objetivo: Personaje) {
        if (hechizo.costoPoder <= poderMagico) {
            objetivo.recibirDanio(hechizo.danio)
            poderMagico -= hechizo.costoPoder
            println(nombre + " lanza " + hechizo.nombre + " a " + objetivo.nombre + " causando " + hechizo.danio + " de daño.")
        } else {
            println(nombre + " no tiene suficiente poder mágico para lanzar " + hechizo.nombre + ".")
        }
    }

    override fun toString(): String {
        return super.toString() + " (Mago, Poder Mágico: " + poderMagico + ")"
    }
}